src/openvic-simulation/research/Technology.cpp

```diff
@@ -114,52 +101,55 @@ bool TechnologyManager::load_technology_file_schools(ModifierManager const& modi
 bool TechnologyManager::load_technologies_file(ModifierManager const& modifier_manager, UnitManager const& unit_manager, BuildingManager const& building_manager, ast::NodeCPtr root) {
 	return expect_dictionary_reserve_length(technologies, [this, &modifier_manager, &unit_manager, &building_manager](std::string_view tech_key, ast::NodeCPtr tech_value) -> bool {
 		ModifierValue modifiers;
-		TechnologyArea const* area = nullptr;
-		Technology::year_t year = 0;
-		fixed_point_t cost = 0;
+		std::string_view area_identifier;
+		Date::year_t year;
+		fixed_point_t cost;
 		bool unciv_military = false;
 		uint8_t unit = 0;
 		Technology::unit_set_t activated_units;
 		Technology::building_set_t activated_buildings;
 
 		bool ret = modifier_manager.expect_modifier_value_and_keys(move_variable_callback(modifiers),
-			"area", ONE_EXACTLY, expect_technology_area_identifier(assign_variable_callback_pointer(area)),
+			"area", ONE_EXACTLY, expect_identifier(assign_variable_callback(area_identifier)),
 			"year", ONE_EXACTLY, expect_uint(assign_variable_callback(year)),
 			"cost", ONE_EXACTLY, expect_fixed_point(assign_variable_callback(cost)),
 			"unciv_military", ZERO_OR_ONE, expect_bool(assign_variable_callback(unciv_military)),
 			"unit", ZERO_OR_ONE, expect_uint(assign_variable_callback(unit)),
-			"activate_unit", ZERO_OR_MORE, unit_manager.expect_unit_identifier([&activated_units](Unit const& unit) -> bool {
+			"activate_unit", ZERO_OR_MORE, unit_manager.expect_unit_identifier([this, &activated_units](Unit const& unit) -> bool {
 				activated_units.insert(&unit);
 				return true;
 			}),
-			"activate_building", ZERO_OR_MORE, building_manager.expect_building_type_identifier(
-				[&activated_buildings](BuildingType const& building_type) -> bool {
-					activated_buildings.insert(&building_type);
-					return true;
-				}
-			),
-			"ai_chance", ONE_EXACTLY, success_callback //TODO
+			"activate_building", ZERO_OR_MORE, building_manager.expect_building_type_identifier([this, &activated_buildings](BuildingType const& type) -> bool {
+				activated_buildings.insert(&type);
+				return true;
+			}),
+			"ai_chance", ONE_EXACTLY, expect_dictionary([this](std::string_view identifier, ast::NodeCPtr value) -> bool { return true; }) //TODO
 		)(tech_value);
 
-		ret &= add_technology(
-			tech_key, area, year, cost, unciv_military, unit, std::move(activated_units), std::move(activated_buildings),
-			std::move(modifiers)
-		);
+		TechnologyArea const* area = get_technology_area_by_identifier(area_identifier);
+
+		ret &= add_technology(tech_key, area, year, cost, unciv_military, unit, activated_units, activated_buildings, std::move(modifiers));
 		return ret;
 	})(root);
 }
 
+#define TECH_MODIFIER(NAME) ret &= modifier_manager.add_modifier_effect(NAME, true, ModifierEffect::format_t::PROPORTION_DECIMAL)
```

Comment:
Again, here you've reverted all the changed I made, can you please reset it to `Technology.cpp` on the current `master` branch and then re-add your changes, which are only here at the end.

ALSO - the unvic modifiers are `positive_good = false`, they affect the cost of westernisation reforms, not the speed of getting points for them.